Sources/App/Features/Presentations/Controllers/PresentationAPIController.swift

```diff
@@ -0,0 +1,75 @@
+import Vapor
+import Fluent
+
+struct PresentationAPIController: RouteCollection {
+    
+    private struct FormInput: Content {
+        let speaker: String
+        let event: String
+        let title: String
+        let synopsis: String
+    }
+    
+    func boot(routes: RoutesBuilder) throws {
+        routes.post("", use: onPost)
+        routes.post(":id", use: onEdit)
+    }
+
+    private func onPost(request: Request) async throws -> Response {
+        let input = try request.content.decode(FormInput.self)
+        
+        guard let speaker = try await Speaker.find(UUID(uuidString: input.speaker), on: request.db) else {
+            return request.redirect(to: "/admin")
+        }
+        
+        guard let event = try await Event.find(.init(uuidString: input.event), on: request.db) else {
+            return request.redirect(to: "/admin")
+        }
+        
+        guard let eventID = event.id else {
+            return request.redirect(to: "/admin")
+        }
+        
+        let presentation = Presentation(
+            id: .generateRandom(),
+            title: input.title,
+            synopsis: input.synopsis,
+            image: nil,
+            eventID: eventID 
+        )
+        
+        presentation.$speaker.id = try speaker.requireID()
+        presentation.$event.id = try event.requireID()
+        
+        try await presentation.create(on: request.db)
+        
+        return request.redirect(to: "/admin")
+    }
+    
+    private func onEdit(request: Request) async throws -> Response {
+        let input = try request.content.decode(FormInput.self)
+        
+        guard let speaker = try await Speaker.find(.init(uuidString: input.speaker), on: request.db) else {
```

Comment:
It could be better to have "UUID" rather than .init as there's quite a lot going on here to take in from the first glance

Reply:
`.init` will always get my love over writing out the type ❤️but no, agreed, I did find it odd that `.find` only accepts a UUID rather than `Identifiable` from Vapor 😕